src/NexusMods.DataModel/Attributes/DiskStateAttribute.cs

```diff
@@ -0,0 +1,60 @@
+using System.Net;
+using System.Runtime.InteropServices;
+using FlatSharp;
+using NexusMods.Abstractions.DiskState;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Archives.Nx.Enums;
+using NexusMods.DataModel.Serializers.DiskStateTreeSchema;
+using NexusMods.Hashing.xxHash64;
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.ElementComparers;
+using NexusMods.Paths;
+using NexusMods.Paths.Trees.Traits;
+using File = NexusMods.DataModel.Serializers.DiskStateTreeSchema.File;
+
+namespace NexusMods.DataModel.Attributes;
+
+public class DiskStateAttribute(string ns, string name) : HashedBlobAttribute<DiskStateTree>(ns, name)
```

Comment:
Doesn't this need to implement `ToLowLevel`?

Reply:
Blobs (and HashedBlobs) are different because what we need at the low level is a `ReadOnlySpan<byte>` to write to the disk. So these expose a lower level interface that hands the attribute a `IBufferedWriter<byte>` that they can use to serialize as they please. I would have prefered it to be a uniform interface, but there's not an easy way to do this without making the normal use case (value types) more complex just to make a rare use case (blobs) simpler.